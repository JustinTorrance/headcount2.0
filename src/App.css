* {
  margin: 0;
  padding: 0;
}

main {
  padding: 0 7px;
}

.title {
  text-align: center;
  margin: 40px 0;
  font-size: 3em;
}
