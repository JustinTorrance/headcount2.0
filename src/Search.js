import React, { Component } from 'react';

class Search extends Component {
  constructor() {
    super();
    this.state = {
      location: ''
    }
  }

  handleChange = (e) => {
    this.setState({location: e.target.value}, 
    () => this.props.searchByLocation(this.state.location)
    )
  }

  render() {
    return(
      <form>
        <p>Search:</p>
        <input 
          type='text' 
          value={this.state.location} 
          name='search' 
          onChange={this.handleChange} 
          placeholder='search by district'/>
      </form>
    )
  }
}

export default Search;