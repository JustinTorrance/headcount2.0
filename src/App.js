import React, { Component } from 'react';
import './App.css';
import CardContainer from './CardContainer.js';
import DistrictRepository from './helper.js';
import kinderData from './data/kindergartners_in_full_day_program.js';
import Search from './Search.js';


class App extends Component {
  constructor() {
    super();
    this.state = {
      SchoolStats: new DistrictRepository(kinderData),
      searchedLocation: ''
    }
  }

  searchByLocation = (searchValue) => {
    this.setState({searchedLocation: searchValue})
  }

  compareAverages = () => {
    const districtsBeingCompared = 
      this.state.districtsBeingCompared;      
    let district1;
    let district2;

    if (districtsBeingCompared.length === 2) {
      district1 = districtsBeingCompared[0].location;
      district2 = districtsBeingCompared[1].location;
      let districtAverages = 
        this.state.data.compareDistrictAverages(district1, district2);
      this.setState({ districtAverages });
    }
  }

  render() {
    let { findAllMatches } = this.state.SchoolStats;
    return (
      <main>
        <h1 className='title'>HeadCount</h1>
        <Search searchByLocation={this.searchByLocation} />
        <CardContainer findAllMatches={findAllMatches} searchedLocation={this.state.searchedLocation} />
      </main>
    );
  }
}

export default App;







  // searchByLocation = (location) => {
  //   const districts = new DistrictRepository(kinderData)
  //   if (location === '') {
  //     return districts
  //   } else {
  //     const matchingDistricts = Object.keys(districts.stats).filter(district => {
  //       return district.includes(location.toUpperCase())
  //     })  
  //     this.setState({searchedLocation: [...matchingDistricts]})
  //   }
  // } 