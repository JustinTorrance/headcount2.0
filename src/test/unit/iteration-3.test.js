import React from 'react';
import {shallow, mount} from 'enzyme';
import DistrictRepository from '../../helper.js';
import kinderData from '../../data/kindergartners_in_full_day_program.js';
import App from '../../App.js';
import Search from '../../Search.js';

describe('Search component iteration 3', () => {
  it('should exist', () => {
     const wrapper = shallow(<Search />);
     expect(wrapper).toBeDefined()
  })
  it('should match snapshot', () => {
    const wrapper = shallow(<Search />)
    expect(wrapper).toMatchSnapshot()
  })
  it('should update state', () => {
    const wrapper = shallow(<Search searchByLocation={jest.fn()}/>)
    const mockedEvent = {
      target: {
        value: 'howdy'
      }
    }
    wrapper.setState({location: ''})
    wrapper.instance().handleChange(mockedEvent)  
    expect(wrapper.state('location')).toEqual('howdy')
  })
  it('should invoke searchByLocaiton', () => {
    const mockedEvent = {
      target: {
        value: 'howdy'
      }
    }
    const mockState = { location: 'howdy' }
    const wrapper = mount(<Search searchByLocation={jest.fn()}/>)
    wrapper.instance().handleChange(mockedEvent)
    expect(wrapper.props().searchByLocation).toHaveBeenCalledWith(mockState.location)
  })
})




