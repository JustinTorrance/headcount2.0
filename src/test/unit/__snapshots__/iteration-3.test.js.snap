// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search component iteration 3 should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Search />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p>
          Search:
</p>,
        <input
          name="search"
          onChange={[Function]}
          placeholder="search by district"
          type="text"
          value=""
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Search:",
        },
        "ref": null,
        "rendered": "Search:",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "name": "search",
          "onChange": [Function],
          "placeholder": "search by district",
          "type": "text",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
    ],
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p>
            Search:
</p>,
          <input
            name="search"
            onChange={[Function]}
            placeholder="search by district"
            type="text"
            value=""
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Search:",
          },
          "ref": null,
          "rendered": "Search:",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "name": "search",
            "onChange": [Function],
            "placeholder": "search by district",
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
      ],
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
